django/contrib/auth/hashers.py

```diff
@@ -384,10 +390,12 @@ def must_update(self, encoded):
         argon2 = self._load_library()
         current_params = argon2.extract_parameters('$' + rest)
         new_params = self.params()
+        # A string of length: N / log_2(26+26+10), provides N bits of entropy.
+        weak_salt = current_params.salt_len < self.salt_entropy / math.log(62, 2)
```

Comment:
While trying to push this PR over the finish line I noted that `salt_len` is not really the `salt_len` we want (it's the size of a string in bytes). I have pushed a new PR #13815 which adds the actual salt to `decode`. I'll update this PR once the other one is merged.